==29872== Memcheck, a memory error detector
==29872== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==29872== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==29872== Command: /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi
==29872== 
--29872-- WARNING: unhandled amd64-linux syscall: 332
--29872-- You may be able to write your own handler.
--29872-- Read the file README_MISSING_SYSCALL_OR_IOCTL.
--29872-- Nevertheless we consider this a bug.  Please report
--29872-- it at http://valgrind.org/support/bug_reports.html.
==29872== Warning: set address range perms: large range [0x59e43000, 0x1d9e43000) (noaccess)
==29872== 
==29872== HEAP SUMMARY:
==29872==     in use at exit: 37,885 bytes in 210 blocks
==29872==   total heap usage: 58,901 allocs, 58,691 frees, 29,887,558 bytes allocated
==29872== 
==29872== 5 bytes in 1 blocks are indirectly lost in loss record 1 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1FE88DB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1FE89A3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1FE712D: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==29872==    by 0x1FE7700: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==29872==    by 0x1FE6EEE: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==29872==    by 0x1FE65CE: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==29872==    by 0x1FE5CC1: alloc::slice::hack::to_vec (slice.rs:157)
==29872==    by 0x1FE587A: alloc::slice::<impl [T]>::to_vec (slice.rs:393)
==29872==    by 0x1CB0E0B: <alloc::vec::Vec<T> as core::clone::Clone>::clone (vec.rs:1940)
==29872==    by 0xCB016F: core::ops::function::FnMut::call_mut (function.rs:154)
==29872==    by 0xE61EA3: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872== 
==29872== 8 bytes in 1 blocks are still reachable in loss record 2 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1189372: new<alloc::sync::Arc<rayon_core::registry::Registry>> (boxed.rs:177)
==29872==    by 0x1189372: rayon_core::util::leak (util.rs:5)
==29872==    by 0x118E66E: rayon_core::registry::set_global_registry::{{closure}}::{{closure}} (registry.rs:196)
==29872==    by 0x11C46A8: core::result::Result<T,E>::map (result.rs:519)
==29872==    by 0x118E709: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872==    by 0x11B1B40: std::sync::once::Once::call_once (once.rs:264)
==29872==    by 0x118E616: rayon_core::registry::set_global_registry (registry.rs:194)
==29872== 
==29872== 9 bytes in 9 blocks are indirectly lost in loss record 3 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1E46BEB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1E46D83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1E3D98D: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==29872==    by 0x1E42810: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==29872==    by 0x1E3C70E: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==29872==    by 0x1E174FE: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==29872==    by 0x1E733D1: alloc::slice::hack::to_vec (slice.rs:157)
==29872==    by 0x1E27F2A: alloc::slice::<impl [T]>::to_vec (slice.rs:393)
==29872==    by 0x1E19ACB: <alloc::vec::Vec<T> as core::clone::Clone>::clone (vec.rs:1940)
==29872==    by 0x1E45B50: <wasmer_types::types::FunctionType as core::clone::Clone>::clone (types.rs:228)
==29872==    by 0x1E55F48: wasmer_vm::sig_registry::SignatureRegistry::register (sig_registry.rs:46)
==29872== 
==29872== 9 bytes in 9 blocks are indirectly lost in loss record 4 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1E46BEB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1E46D83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1E3D98D: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==29872==    by 0x1E42810: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==29872==    by 0x1E3C70E: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==29872==    by 0x1E174FE: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==29872==    by 0x1E733D1: alloc::slice::hack::to_vec (slice.rs:157)
==29872==    by 0x1E27F2A: alloc::slice::<impl [T]>::to_vec (slice.rs:393)
==29872==    by 0x1E19ACB: <alloc::vec::Vec<T> as core::clone::Clone>::clone (vec.rs:1940)
==29872==    by 0x1E45B50: <wasmer_types::types::FunctionType as core::clone::Clone>::clone (types.rs:228)
==29872==    by 0x1E563ED: wasmer_vm::sig_registry::SignatureRegistry::register (sig_registry.rs:58)
==29872== 
==29872== 16 bytes in 1 blocks are indirectly lost in loss record 5 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1F20D6B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1F21513: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1F1FE3A: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==29872==    by 0x1CC0FB0: alloc::raw_vec::RawVec<T,A>::grow_amortized (raw_vec.rs:429)
==29872==    by 0x1CBA313: alloc::raw_vec::RawVec<T,A>::try_reserve (raw_vec.rs:310)
==29872==    by 0x1CC4292: alloc::raw_vec::RawVec<T,A>::reserve (raw_vec.rs:300)
==29872==    by 0x1CB0348: alloc::vec::Vec<T>::reserve (vec.rs:505)
==29872==    by 0xE3EDB4: alloc::vec::Vec<T>::push (vec.rs:1206)
==29872==    by 0xF14B13: wasmer_wasi::state::WasiFs::new_init (mod.rs:460)
==29872==    by 0xF11F8F: wasmer_wasi::state::WasiFs::new_with_preopen (mod.rs:294)
==29872==    by 0xE11859: wasmer_wasi::state::builder::WasiStateBuilder::build (builder.rs:366)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 6 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE96D36: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE96D36: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE66CD9: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:206)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 7 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE988B6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE988B6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE66D5A: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:207)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 8 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE96FF6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE96FF6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE66DCB: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:208)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 9 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE964F6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE964F6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE66E3C: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:209)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 10 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE96236: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE96236: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE66EAD: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:210)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 11 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9BA36: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9BA36: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE66F1E: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:211)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 12 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9A176: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9A176: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE66F8F: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:212)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 13 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9C276: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9C276: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67071: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:214)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 14 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE99EB6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE99EB6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE670E2: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:215)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 15 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9B776: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9B776: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67153: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:216)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 16 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE990F6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE990F6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE671C4: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:217)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 17 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9D876: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9D876: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67235: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:218)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 18 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9DDF6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9DDF6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE672A6: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:219)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 19 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE98E36: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE98E36: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67317: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:220)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 20 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE96A76: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE96A76: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67388: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:221)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 21 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9D2F6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9D2F6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE673F9: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:222)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 22 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9BFB6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9BFB6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE6746A: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:223)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 23 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE98076: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE98076: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE674DB: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:224)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 24 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9A6F6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9A6F6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE6754C: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:225)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 25 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9D5B6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9D5B6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE675BD: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:226)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 26 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE967B6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE967B6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE6762E: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:227)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 27 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE972B6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE972B6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE6769F: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:228)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 28 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9DB36: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9DB36: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67710: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:229)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 29 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9C7F6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9C7F6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67781: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:230)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 30 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9A9B6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9A9B6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE677F2: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:231)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 31 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9CD76: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9CD76: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67863: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:232)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 32 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9BCF6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9BCF6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE678D4: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:233)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 33 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE98336: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE98336: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67945: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:234)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 34 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE993B6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE993B6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE679B6: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:235)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 35 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9D036: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9D036: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67A27: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:236)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 36 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE985F6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE985F6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67A98: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:237)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 37 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE97DB6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE97DB6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67B09: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:238)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 38 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9E0B6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9E0B6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67B7A: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:239)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 39 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE99676: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE99676: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67BEB: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:240)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 40 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE97576: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE97576: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67C5C: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:241)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 41 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE99936: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE99936: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67CCD: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:242)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 42 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9E636: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9E636: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67D3E: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:243)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 43 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9A436: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9A436: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67DAF: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:244)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 44 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE97836: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE97836: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67E14: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:245)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 45 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9CAB6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9CAB6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67E79: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:246)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 46 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9E376: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9E376: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67EDE: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:247)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 47 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9AC76: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9AC76: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67F43: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:248)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 48 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9C536: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9C536: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67FA8: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:249)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 16 bytes in 1 blocks are definitely lost in loss record 49 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9B4B6: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9B4B6: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE6800D: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:250)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 18 bytes in 4 blocks are indirectly lost in loss record 50 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1FE88DB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1FE89A3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1FE712D: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==29872==    by 0x1FE7700: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==29872==    by 0x1FE6EEE: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==29872==    by 0x1FE65CE: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==29872==    by 0x1FE5CC1: alloc::slice::hack::to_vec (slice.rs:157)
==29872==    by 0x1FE587A: alloc::slice::<impl [T]>::to_vec (slice.rs:393)
==29872==    by 0x1FE58AA: alloc::slice::<impl alloc::borrow::ToOwned for [T]>::to_owned (slice.rs:726)
==29872==    by 0xEB49B5: alloc::str::<impl alloc::borrow::ToOwned for str>::to_owned (str.rs:205)
==29872==    by 0xE1C679: <alloc::string::String as core::convert::From<&str>>::from (string.rs:2309)
==29872== 
==29872== 24 bytes in 1 blocks are indirectly lost in loss record 51 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xDDB4ED: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==29872==    by 0xDE1970: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==29872==    by 0xDD9ABE: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==29872==    by 0xE3BBFE: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==29872==    by 0xE5D801: alloc::slice::hack::to_vec (slice.rs:157)
==29872==    by 0xE4030A: alloc::slice::<impl [T]>::to_vec (slice.rs:393)
==29872==    by 0xE4094B: <alloc::vec::Vec<T> as core::clone::Clone>::clone (vec.rs:1940)
==29872==    by 0xE120B9: wasmer_wasi::state::builder::WasiStateBuilder::build (builder.rs:389)
==29872==    by 0xE12544: wasmer_wasi::state::builder::WasiStateBuilder::finalize (builder.rs:409)
==29872== 
==29872== 32 bytes in 1 blocks are still reachable in loss record 52 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE6539B: new<inventory::Node<wasmer_wasi::state::types::_WasiFile_registry::TypetagRegistration>> (boxed.rs:177)
==29872==    by 0xE6539B: inventory::submit (lib.rs:160)
==29872==    by 0xCA6ABB: wasmer_wasi::state::types::__init10095839713170728602::__init10095839713170728602 (types.rs:566)
==29872==    by 0x20C949C: __libc_csu_init (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872==    by 0x56A2B87: (below main) (libc-start.c:266)
==29872== 
==29872== 32 bytes in 1 blocks are still reachable in loss record 53 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE6539B: new<inventory::Node<wasmer_wasi::state::types::_WasiFile_registry::TypetagRegistration>> (boxed.rs:177)
==29872==    by 0xE6539B: inventory::submit (lib.rs:160)
==29872==    by 0xCA6AFB: wasmer_wasi::state::types::__init17928626087104593939::__init17928626087104593939 (types.rs:743)
==29872==    by 0x20C949C: __libc_csu_init (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872==    by 0x56A2B87: (below main) (libc-start.c:266)
==29872== 
==29872== 32 bytes in 1 blocks are still reachable in loss record 54 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE6539B: new<inventory::Node<wasmer_wasi::state::types::_WasiFile_registry::TypetagRegistration>> (boxed.rs:177)
==29872==    by 0xE6539B: inventory::submit (lib.rs:160)
==29872==    by 0xCA6B3B: wasmer_wasi::state::types::__init18259907151022521602::__init18259907151022521602 (types.rs:836)
==29872==    by 0x20C949C: __libc_csu_init (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872==    by 0x56A2B87: (below main) (libc-start.c:266)
==29872== 
==29872== 32 bytes in 1 blocks are still reachable in loss record 55 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE6539B: new<inventory::Node<wasmer_wasi::state::types::_WasiFile_registry::TypetagRegistration>> (boxed.rs:177)
==29872==    by 0xE6539B: inventory::submit (lib.rs:160)
==29872==    by 0xCA6B7B: wasmer_wasi::state::types::__init13015983451510593879::__init13015983451510593879 (types.rs:929)
==29872==    by 0x20C949C: __libc_csu_init (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872==    by 0x56A2B87: (below main) (libc-start.c:266)
==29872== 
==29872== 32 bytes in 1 blocks are still reachable in loss record 56 of 107
==29872==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==29872==    by 0x4E3F165: dlsym (dlsym.c:70)
==29872==    by 0x2084A73: fetch (weak.rs:61)
==29872==    by 0x2084A73: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==29872==    by 0x2084A73: min_stack_size (thread.rs:449)
==29872==    by 0x2084A73: std::sys::unix::thread::Thread::new (thread.rs:49)
==29872==    by 0x11BE0AB: std::thread::Builder::spawn_unchecked (mod.rs:492)
==29872==    by 0x11BF2DB: std::thread::Builder::spawn (mod.rs:386)
==29872==    by 0x118E0F8: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29872==    by 0x118F1DF: rayon_core::registry::Registry::new (registry.rs:260)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872== 
==29872== 32 bytes in 1 blocks are indirectly lost in loss record 57 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x10CBA3B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x10CBBD3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x10CB99C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x10901B7: new<wasmer_compiler_cranelift::compiler::CraneliftCompiler> (boxed.rs:177)
==29872==    by 0x10901B7: <wasmer_compiler_cranelift::config::Cranelift as wasmer_compiler::compiler::CompilerConfig>::compiler (config.rs:198)
==29872==    by 0x1033DD2: wasmer_engine_jit::builder::JIT::engine (builder.rs:51)
==29872==    by 0xCAA8AE: wasi::main (wasi.rs:35)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872==    by 0xCAB5E9: main (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872== 
==29872== 32 bytes in 1 blocks are indirectly lost in loss record 58 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1F4DA5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1F4DE93: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1F5104A: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==29872==    by 0x1F53AF0: alloc::raw_vec::RawVec<T,A>::grow_amortized (raw_vec.rs:429)
==29872==    by 0x1F51A73: alloc::raw_vec::RawVec<T,A>::try_reserve (raw_vec.rs:310)
==29872==    by 0x1F541E2: alloc::raw_vec::RawVec<T,A>::reserve (raw_vec.rs:300)
==29872==    by 0x1F63398: alloc::vec::Vec<T>::reserve (vec.rs:505)
==29872==    by 0xFE3E36: alloc::vec::Vec<T>::push (vec.rs:1206)
==29872==    by 0x1034889: wasmer_engine_jit::unwind::systemv::UnwindRegistry::register_frames (systemv.rs:73)
==29872==    by 0x10345FF: wasmer_engine_jit::unwind::systemv::UnwindRegistry::publish (systemv.rs:51)
==29872==    by 0xFB2629: wasmer_engine_jit::engine::JITEngineInner::publish_eh_frame (engine.rs:285)
==29872== 
==29872== 40 bytes in 1 blocks are indirectly lost in loss record 59 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xFA068B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xFA0823: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xFA05EC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1008C4D: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29872==    by 0xFB0A73: wasmer_engine_jit::engine::JITEngine::new (engine.rs:33)
==29872==    by 0x1033E74: wasmer_engine_jit::builder::JIT::engine (builder.rs:52)
==29872==    by 0xCAA8AE: wasi::main (wasi.rs:35)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872==    by 0xCAB5E9: main (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872== 
==29872== 40 bytes in 1 blocks are indirectly lost in loss record 60 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xFA068B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xFA0823: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xFA05EC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xFEC7B7: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0xFB156A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::cloned (engine.rs:136)
==29872==    by 0xCAA043: wasmer::store::Store::new (store.rs:32)
==29872==    by 0xCAA8D5: wasi::main (wasi.rs:35)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872==    by 0xCAB5E9: main (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872== 
==29872== 40 bytes in 1 blocks are indirectly lost in loss record 61 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xCA8B6B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xCA8C93: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xCA8ACC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xCADAE7: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0xCAA0B0: wasmer::store::Store::new (store.rs:33)
==29872==    by 0xCAA8D5: wasi::main (wasi.rs:35)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872==    by 0xCAB5E9: main (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872== 
==29872== 40 bytes in 1 blocks are indirectly lost in loss record 62 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xDFA1DD: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29872==    by 0xE6671D: wasmer_wasi::WasiEnv::new (lib.rs:142)
==29872==    by 0xE12628: wasmer_wasi::state::builder::WasiStateBuilder::finalize (builder.rs:410)
==29872==    by 0xCAABB0: wasi::main (wasi.rs:43)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872==    by 0xCAB5E9: main (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872== 
==29872== 40 bytes in 1 blocks are indirectly lost in loss record 63 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1EA9B1B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1EAAEF3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1EA9A7C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1EDAF6C: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29872==    by 0xE66437: wasmer_wasi::WasiMemory::new (lib.rs:81)
==29872==    by 0xE6674B: wasmer_wasi::WasiEnv::new (lib.rs:143)
==29872==    by 0xE12628: wasmer_wasi::state::builder::WasiStateBuilder::finalize (builder.rs:410)
==29872==    by 0xCAABB0: wasi::main (wasi.rs:43)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 40 bytes in 1 blocks are indirectly lost in loss record 64 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1E46BEB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1E46D83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1E46B4C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1E38527: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29872==    by 0x1E0EAC9: wasmer_vm::memory::LinearMemory::new_internal (memory.rs:264)
==29872==    by 0x1E0E188: wasmer_vm::memory::LinearMemory::from_definition (memory.rs:199)
==29872==    by 0x1A324EB: <wasmer::tunables::Tunables as wasmer_engine::tunables::Tunables>::create_vm_memory (tunables.rs:110)
==29872==    by 0xCA81DF: wasmer_engine::tunables::Tunables::create_memories (tunables.rs:80)
==29872==    by 0x104EFE5: wasmer_engine::artifact::Artifact::instantiate (artifact.rs:110)
==29872==    by 0x1A0D2BD: wasmer::module::Module::instantiate (module.rs:267)
==29872==    by 0x1A26B2D: wasmer::instance::Instance::new (instance.rs:77)
==29872== 
==29872== 42 bytes in 13 blocks are indirectly lost in loss record 65 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1E46BEB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1E46D83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1E3D98D: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==29872==    by 0x1E42810: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==29872==    by 0x1E3C70E: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==29872==    by 0x1E174FE: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==29872==    by 0x1E733D1: alloc::slice::hack::to_vec (slice.rs:157)
==29872==    by 0x1E27F2A: alloc::slice::<impl [T]>::to_vec (slice.rs:393)
==29872==    by 0x1E19ACB: <alloc::vec::Vec<T> as core::clone::Clone>::clone (vec.rs:1940)
==29872==    by 0x1E45B2F: <wasmer_types::types::FunctionType as core::clone::Clone>::clone (types.rs:226)
==29872==    by 0x1E55F48: wasmer_vm::sig_registry::SignatureRegistry::register (sig_registry.rs:46)
==29872== 
==29872== 42 bytes in 13 blocks are indirectly lost in loss record 66 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1E46BEB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1E46D83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1E3D98D: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==29872==    by 0x1E42810: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==29872==    by 0x1E3C70E: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==29872==    by 0x1E174FE: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==29872==    by 0x1E733D1: alloc::slice::hack::to_vec (slice.rs:157)
==29872==    by 0x1E27F2A: alloc::slice::<impl [T]>::to_vec (slice.rs:393)
==29872==    by 0x1E19ACB: <alloc::vec::Vec<T> as core::clone::Clone>::clone (vec.rs:1940)
==29872==    by 0x1E45B2F: <wasmer_types::types::FunctionType as core::clone::Clone>::clone (types.rs:226)
==29872==    by 0x1E563ED: wasmer_vm::sig_registry::SignatureRegistry::register (sig_registry.rs:58)
==29872== 
==29872== 48 bytes in 1 blocks are still reachable in loss record 67 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x547B6ED: __register_frame (in /lib/x86_64-linux-gnu/libgcc_s.so.1)
==29872==    by 0x1034876: wasmer_engine_jit::unwind::systemv::UnwindRegistry::register_frames (systemv.rs:72)
==29872==    by 0x10345FF: wasmer_engine_jit::unwind::systemv::UnwindRegistry::publish (systemv.rs:51)
==29872==    by 0xFB2629: wasmer_engine_jit::engine::JITEngineInner::publish_eh_frame (engine.rs:285)
==29872==    by 0x102EA95: wasmer_engine_jit::artifact::JITArtifact::from_parts (artifact.rs:209)
==29872==    by 0x102DA04: wasmer_engine_jit::artifact::JITArtifact::new (artifact.rs:124)
==29872==    by 0xFB119E: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (engine.rs:110)
==29872==    by 0x1A0CCA2: wasmer::module::Module::compile (module.rs:161)
==29872==    by 0x1A0C9DB: wasmer::module::Module::from_binary_unchecked (module.rs:146)
==29872==    by 0x1A0C8E8: wasmer::module::Module::from_binary (module.rs:132)
==29872==    by 0xCADF12: wasmer::module::Module::new (module.rs:109)
==29872== 
==29872== 64 bytes in 4 blocks are still reachable in loss record 68 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x119D056: new<crossbeam_deque::deque::Buffer<rayon_core::job::JobRef>> (boxed.rs:177)
==29872==    by 0x119D056: <T as crossbeam_epoch::atomic::Pointable>::init (atomic.rs:188)
==29872==    by 0x11B9903: crossbeam_epoch::atomic::Owned<T>::init (atomic.rs:861)
==29872==    by 0x11B9A53: crossbeam_epoch::atomic::Atomic<T>::init (atomic.rs:309)
==29872==    by 0x11B9A23: crossbeam_epoch::atomic::Atomic<T>::new (atomic.rs:294)
==29872==    by 0x119B756: crossbeam_deque::deque::Worker<T>::new_lifo (deque.rs:253)
==29872==    by 0x118F65E: rayon_core::registry::Registry::new::{{closure}} (registry.rs:229)
==29872==    by 0x11C836C: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872==    by 0x1198FBD: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29872== 
==29872== 64 bytes in 4 blocks are still reachable in loss record 69 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x20849CC: alloc (alloc.rs:80)
==29872==    by 0x20849CC: alloc (alloc.rs:174)
==29872==    by 0x20849CC: exchange_malloc (alloc.rs:277)
==29872==    by 0x20849CC: std::sys::unix::thread::Thread::new (thread.rs:44)
==29872==    by 0x11BE0AB: std::thread::Builder::spawn_unchecked (mod.rs:492)
==29872==    by 0x11BF2DB: std::thread::Builder::spawn (mod.rs:386)
==29872==    by 0x118E0F8: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29872==    by 0x118F1DF: rayon_core::registry::Registry::new (registry.rs:260)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872==    by 0x11B1B40: std::sync::once::Once::call_once (once.rs:264)
==29872==    by 0x118E616: rayon_core::registry::set_global_registry (registry.rs:194)
==29872== 
==29872== 64 bytes in 1 blocks are indirectly lost in loss record 70 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xFA068B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xFA0823: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xFA05EC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xFEC26D: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0xFB0AE8: wasmer_engine_jit::engine::JITEngine::new (engine.rs:39)
==29872==    by 0x1033E74: wasmer_engine_jit::builder::JIT::engine (builder.rs:52)
==29872==    by 0xCAA8AE: wasi::main (wasi.rs:35)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872==    by 0xCAB5E9: main (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872== 
==29872== 72 bytes in 1 blocks are still reachable in loss record 71 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1C76D9B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1C76EC3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1C76CFC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1C79D17: std::sync::rwlock::RwLock<T>::new (rwlock.rs:134)
==29872==    by 0x1C7A04C: <std::sync::rwlock::RwLock<T> as core::default::Default>::default (rwlock.rs:451)
==29872==    by 0x1C6B820: __static_ref_initialize (frame_info.rs:28)
==29872==    by 0x1C6B820: core::ops::function::FnOnce::call_once (function.rs:233)
==29872==    by 0x1CA31EB: lazy_static::lazy::Lazy<T>::get::{{closure}} (inline_lazy.rs:31)
==29872==    by 0x1CA1B64: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872==    by 0x1CA1AE0: std::sync::once::Once::call_once (once.rs:264)
==29872==    by 0x1C66C48: get<std::sync::rwlock::RwLock<wasmer_engine::trap::frame_info::GlobalFrameInfo>,fn() -> std::sync::rwlock::RwLock<wasmer_engine::trap::frame_info::GlobalFrameInfo>> (inline_lazy.rs:30)
==29872==    by 0x1C66C48: __stability (lib.rs:142)
==29872==    by 0x1C66C48: <wasmer_engine::trap::frame_info::FRAME_INFO as core::ops::deref::Deref>::deref (lib.rs:144)
==29872== 
==29872== 72 bytes in 1 blocks are indirectly lost in loss record 72 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1E46BEB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1E46D83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1E46B4C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1E73A1D: std::sync::rwlock::RwLock<T>::new (rwlock.rs:134)
==29872==    by 0x1E74250: <std::sync::rwlock::RwLock<T> as core::default::Default>::default (rwlock.rs:451)
==29872==    by 0x1E55E01: wasmer_vm::sig_registry::SignatureRegistry::new (sig_registry.rs:38)
==29872==    by 0xFB0963: wasmer_engine_jit::engine::JITEngine::new (engine.rs:36)
==29872==    by 0x1033E74: wasmer_engine_jit::builder::JIT::engine (builder.rs:52)
==29872==    by 0xCAA8AE: wasi::main (wasi.rs:35)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872== 
==29872== 88 bytes in 1 blocks are indirectly lost in loss record 73 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE4715D: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0xE6676B: wasmer_wasi::WasiEnv::new (lib.rs:143)
==29872==    by 0xE12628: wasmer_wasi::state::builder::WasiStateBuilder::finalize (builder.rs:410)
==29872==    by 0xCAABB0: wasi::main (wasi.rs:43)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872==    by 0xCAB5E9: main (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872== 
==29872== 128 bytes in 4 blocks are still reachable in loss record 74 of 107
==29872==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==29872==    by 0x2050FFE: register_dtor (fast_thread_local.rs:33)
==29872==    by 0x2050FFE: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==29872==    by 0x2050FFE: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==29872==    by 0x20737F9: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==29872==    by 0x20737F9: __getit (local.rs:175)
==29872==    by 0x20737F9: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==29872==    by 0x20737F9: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==29872==    by 0x20737F9: std::sys_common::thread_info::set (thread_info.rs:40)
==29872==    by 0x11BE471: std::thread::Builder::spawn_unchecked::{{closure}} (mod.rs:473)
==29872==    by 0x11B2B8E: core::ops::function::FnOnce::call_once{{vtable-shim}} (function.rs:233)
==29872==    by 0x2084DE9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==29872==    by 0x2084DE9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==29872==    by 0x2084DE9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==29872==    by 0x52516DA: start_thread (pthread_create.c:463)
==29872==    by 0x57A271E: clone (clone.S:95)
==29872== 
==29872== 128 bytes in 4 blocks are still reachable in loss record 75 of 107
==29872==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==29872==    by 0x11F63E1: std::thread::local::fast::Key<T>::try_register_dtor (local.rs:438)
==29872==    by 0x11F62E2: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==29872==    by 0x11F6486: std::thread::local::fast::Key<T>::get (local.rs:410)
==29872==    by 0x11F61DC: crossbeam_epoch::default::HANDLE::__getit (local.rs:175)
==29872==    by 0x119FE84: std::thread::local::LocalKey<T>::try_with (local.rs:262)
==29872==    by 0x11A5812: crossbeam_epoch::default::with_handle (default.rs:43)
==29872==    by 0x11A5AF6: crossbeam_epoch::default::is_pinned (default.rs:30)
==29872==    by 0x119B91C: crossbeam_deque::deque::Stealer<T>::steal (deque.rs:619)
==29872==    by 0x11914C2: rayon_core::registry::WorkerThread::steal::{{closure}} (registry.rs:779)
==29872==    by 0x118C8AC: core::iter::traits::iterator::Iterator::find_map::check::{{closure}} (iterator.rs:2258)
==29872== 
==29872== 128 bytes in 1 blocks are indirectly lost in loss record 76 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1A2B2BB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1A2B453: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1A2B21C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1A2CCFD: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0x1A32603: <wasmer::tunables::Tunables as wasmer_engine::tunables::Tunables>::create_vm_memory (tunables.rs:110)
==29872==    by 0xCA81DF: wasmer_engine::tunables::Tunables::create_memories (tunables.rs:80)
==29872==    by 0x104EFE5: wasmer_engine::artifact::Artifact::instantiate (artifact.rs:110)
==29872==    by 0x1A0D2BD: wasmer::module::Module::instantiate (module.rs:267)
==29872==    by 0x1A26B2D: wasmer::instance::Instance::new (instance.rs:77)
==29872==    by 0xCAADDB: wasi::main (wasi.rs:52)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872== 
==29872== 160 bytes in 4 blocks are still reachable in loss record 77 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x11BC7BF: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29872==    by 0x11CFFD0: rayon_core::latch::LockLatch::new (latch.rs:230)
==29872==    by 0x119040C: rayon_core::registry::ThreadInfo::new (registry.rs:583)
==29872==    by 0x11B2AA1: core::ops::function::FnMut::call_mut (function.rs:154)
==29872==    by 0x11C8083: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872==    by 0x11ABA23: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29872==    by 0x11C9938: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29872==    by 0x11C7839: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29872== 
==29872== 160 bytes in 4 blocks are still reachable in loss record 78 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x11BC7BF: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29872==    by 0x11CFFD0: rayon_core::latch::LockLatch::new (latch.rs:230)
==29872==    by 0x1190424: rayon_core::registry::ThreadInfo::new (registry.rs:584)
==29872==    by 0x11B2AA1: core::ops::function::FnMut::call_mut (function.rs:154)
==29872==    by 0x11C8083: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872==    by 0x11ABA23: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29872==    by 0x11C9938: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29872==    by 0x11C7839: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29872== 
==29872== 160 bytes in 4 blocks are still reachable in loss record 79 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x11BC7BF: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29872==    by 0x11C0133: <std::sync::mutex::Mutex<T> as core::default::Default>::default (mutex.rs:446)
==29872==    by 0x11AAE1E: <rayon_core::sleep::WorkerSleepState as core::default::Default>::default (mod.rs:55)
==29872==    by 0x11A824D: <crossbeam_utils::cache_padded::CachePadded<T> as core::default::Default>::default (cache_padded.rs:77)
==29872==    by 0x11A96BA: rayon_core::sleep::Sleep::new::{{closure}} (mod.rs:67)
==29872==    by 0x11C8230: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872==    by 0x1198DC8: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29872==    by 0x11C9A09: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29872== 
==29872== 160 bytes in 4 blocks are still reachable in loss record 80 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x205239E: alloc (alloc.rs:80)
==29872==    by 0x205239E: alloc (alloc.rs:174)
==29872==    by 0x205239E: exchange_malloc (alloc.rs:277)
==29872==    by 0x205239E: new<()> (mutex.rs:223)
==29872==    by 0x205239E: std::thread::Thread::new (mod.rs:1149)
==29872==    by 0x11BDE4D: std::thread::Builder::spawn_unchecked (mod.rs:462)
==29872==    by 0x11BF2DB: std::thread::Builder::spawn (mod.rs:386)
==29872==    by 0x118E0F8: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29872==    by 0x118F1DF: rayon_core::registry::Registry::new (registry.rs:260)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872==    by 0x11B1B40: std::sync::once::Once::call_once (once.rs:264)
==29872==    by 0x118E616: rayon_core::registry::set_global_registry (registry.rs:194)
==29872== 
==29872== 160 bytes in 4 blocks are still reachable in loss record 81 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1187817: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0x11BDEEB: std::thread::Builder::spawn_unchecked (mod.rs:465)
==29872==    by 0x11BF2DB: std::thread::Builder::spawn (mod.rs:386)
==29872==    by 0x118E0F8: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29872==    by 0x118F1DF: rayon_core::registry::Registry::new (registry.rs:260)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872== 
==29872== 192 bytes in 4 blocks are still reachable in loss record 82 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x206F496: alloc (alloc.rs:80)
==29872==    by 0x206F496: alloc (alloc.rs:174)
==29872==    by 0x206F496: exchange_malloc (alloc.rs:277)
==29872==    by 0x206F496: std::sync::condvar::Condvar::new (condvar.rs:130)
==29872==    by 0x11CFFEC: rayon_core::latch::LockLatch::new (latch.rs:231)
==29872==    by 0x119040C: rayon_core::registry::ThreadInfo::new (registry.rs:583)
==29872==    by 0x11B2AA1: core::ops::function::FnMut::call_mut (function.rs:154)
==29872==    by 0x11C8083: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872==    by 0x11ABA23: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29872==    by 0x11C9938: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29872==    by 0x11C7839: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29872==    by 0x11AF411: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2140)
==29872==    by 0x11B0094: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29872==    by 0x11B1036: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==29872== 
==29872== 192 bytes in 4 blocks are still reachable in loss record 83 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x206F496: alloc (alloc.rs:80)
==29872==    by 0x206F496: alloc (alloc.rs:174)
==29872==    by 0x206F496: exchange_malloc (alloc.rs:277)
==29872==    by 0x206F496: std::sync::condvar::Condvar::new (condvar.rs:130)
==29872==    by 0x11CFFEC: rayon_core::latch::LockLatch::new (latch.rs:231)
==29872==    by 0x1190424: rayon_core::registry::ThreadInfo::new (registry.rs:584)
==29872==    by 0x11B2AA1: core::ops::function::FnMut::call_mut (function.rs:154)
==29872==    by 0x11C8083: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872==    by 0x11ABA23: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29872==    by 0x11C9938: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29872==    by 0x11C7839: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29872==    by 0x11AF411: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2140)
==29872==    by 0x11B0094: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29872==    by 0x11B1036: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==29872== 
==29872== 192 bytes in 4 blocks are still reachable in loss record 84 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x206F496: alloc (alloc.rs:80)
==29872==    by 0x206F496: alloc (alloc.rs:174)
==29872==    by 0x206F496: exchange_malloc (alloc.rs:277)
==29872==    by 0x206F496: std::sync::condvar::Condvar::new (condvar.rs:130)
==29872==    by 0x11AAE3A: <rayon_core::sleep::WorkerSleepState as core::default::Default>::default (mod.rs:57)
==29872==    by 0x11A824D: <crossbeam_utils::cache_padded::CachePadded<T> as core::default::Default>::default (cache_padded.rs:77)
==29872==    by 0x11A96BA: rayon_core::sleep::Sleep::new::{{closure}} (mod.rs:67)
==29872==    by 0x11C8230: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872==    by 0x1198DC8: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29872==    by 0x11C9A09: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29872==    by 0x11C766A: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29872==    by 0x11AF7D7: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2140)
==29872==    by 0x11AFFF1: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29872==    by 0x11B0FD5: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==29872== 
==29872== 192 bytes in 4 blocks are still reachable in loss record 85 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x2052402: alloc (alloc.rs:80)
==29872==    by 0x2052402: alloc (alloc.rs:174)
==29872==    by 0x2052402: exchange_malloc (alloc.rs:277)
==29872==    by 0x2052402: new (condvar.rs:130)
==29872==    by 0x2052402: std::thread::Thread::new (mod.rs:1150)
==29872==    by 0x11BDE4D: std::thread::Builder::spawn_unchecked (mod.rs:462)
==29872==    by 0x11BF2DB: std::thread::Builder::spawn (mod.rs:386)
==29872==    by 0x118E0F8: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29872==    by 0x118F1DF: rayon_core::registry::Registry::new (registry.rs:260)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872==    by 0x11B1B40: std::sync::once::Once::call_once (once.rs:264)
==29872==    by 0x118E616: rayon_core::registry::set_global_registry (registry.rs:194)
==29872== 
==29872== 216 bytes in 1 blocks are indirectly lost in loss record 86 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xFA068B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xFA0823: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xFA05EC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xFEC41D: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0xFB0A96: wasmer_engine_jit::engine::JITEngine::new (engine.rs:33)
==29872==    by 0x1033E74: wasmer_engine_jit::builder::JIT::engine (builder.rs:52)
==29872==    by 0xCAA8AE: wasi::main (wasi.rs:35)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872==    by 0xCAB5E9: main (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872== 
==29872== 224 bytes in 1 blocks are indirectly lost in loss record 87 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1F20D6B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1F21513: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1F1FE3A: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==29872==    by 0xFA8B50: alloc::raw_vec::RawVec<T,A>::grow_amortized (raw_vec.rs:429)
==29872==    by 0xFA4FD3: alloc::raw_vec::RawVec<T,A>::try_reserve (raw_vec.rs:310)
==29872==    by 0xFADF52: alloc::raw_vec::RawVec<T,A>::reserve (raw_vec.rs:300)
==29872==    by 0xFE5C48: alloc::vec::Vec<T>::reserve (vec.rs:505)
==29872==    by 0xFE4C13: alloc::vec::Vec<T>::push (vec.rs:1206)
==29872==    by 0xFB1A9D: wasmer_engine_jit::engine::JITEngineInner::allocate (engine.rs:211)
==29872==    by 0x102E4A9: wasmer_engine_jit::artifact::JITArtifact::from_parts (artifact.rs:164)
==29872==    by 0x102DA04: wasmer_engine_jit::artifact::JITArtifact::new (artifact.rs:124)
==29872== 
==29872== 320 bytes in 4 blocks are still reachable in loss record 88 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x2052451: alloc (alloc.rs:80)
==29872==    by 0x2052451: alloc (alloc.rs:174)
==29872==    by 0x2052451: exchange_malloc (alloc.rs:277)
==29872==    by 0x2052451: new<std::thread::Inner> (sync.rs:324)
==29872==    by 0x2052451: std::thread::Thread::new (mod.rs:1145)
==29872==    by 0x11BDE4D: std::thread::Builder::spawn_unchecked (mod.rs:462)
==29872==    by 0x11BF2DB: std::thread::Builder::spawn (mod.rs:386)
==29872==    by 0x118E0F8: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29872==    by 0x118F1DF: rayon_core::registry::Registry::new (registry.rs:260)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872==    by 0x11B1B40: std::sync::once::Once::call_once (once.rs:264)
==29872==    by 0x118E616: rayon_core::registry::set_global_registry (registry.rs:194)
==29872== 
==29872== 344 bytes in 1 blocks are indirectly lost in loss record 89 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE4733D: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0xE66739: wasmer_wasi::WasiEnv::new (lib.rs:142)
==29872==    by 0xE12628: wasmer_wasi::state::builder::WasiStateBuilder::finalize (builder.rs:410)
==29872==    by 0xCAABB0: wasi::main (wasi.rs:43)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872==    by 0xCAB5E9: main (in /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/wasi)
==29872== 
==29872== 384 bytes in 1 blocks are still reachable in loss record 90 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x121ED4B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x121F5F3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x121610A: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==29872==    by 0x11D2D90: alloc::raw_vec::RawVec<T,A>::grow_amortized (raw_vec.rs:429)
==29872==    by 0x11D16E3: alloc::raw_vec::RawVec<T,A>::try_reserve (raw_vec.rs:310)
==29872==    by 0x11D52E2: alloc::raw_vec::RawVec<T,A>::reserve (raw_vec.rs:300)
==29872==    by 0x11AE1C8: alloc::vec::Vec<T>::reserve (vec.rs:505)
==29872==    by 0x11AF327: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2136)
==29872==    by 0x11B0094: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29872==    by 0x11B1036: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==29872==    by 0x11C7498: core::iter::traits::iterator::Iterator::collect (iterator.rs:1671)
==29872== 
==29872== 416 bytes in 4 blocks are still reachable in loss record 91 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x11BE017: new<closure-0> (boxed.rs:177)
==29872==    by 0x11BE017: std::thread::Builder::spawn_unchecked (mod.rs:494)
==29872==    by 0x11BF2DB: std::thread::Builder::spawn (mod.rs:386)
==29872==    by 0x118E0F8: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29872==    by 0x118F1DF: rayon_core::registry::Registry::new (registry.rs:260)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872== 
==29872== 472 bytes in 1 blocks are indirectly lost in loss record 92 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE87D50: hashbrown::raw::RawTable<T>::new_uninitialized (mod.rs:383)
==29872==    by 0xE885E8: hashbrown::raw::RawTable<T>::try_with_capacity (mod.rs:407)
==29872==    by 0xE8DA42: hashbrown::raw::RawTable<T>::resize (mod.rs:788)
==29872==    by 0xE835CC: hashbrown::raw::RawTable<T>::reserve_rehash (mod.rs:669)
==29872==    by 0xE8E1F6: hashbrown::raw::RawTable<T>::reserve (mod.rs:626)
==29872==    by 0xE8C1F5: hashbrown::raw::RawTable<T>::insert (mod.rs:841)
==29872==    by 0xE1F927: hashbrown::map::HashMap<K,V,S>::insert (map.rs:855)
==29872==    by 0xEB86E9: std::collections::hash::map::HashMap<K,V,S>::insert (map.rs:819)
==29872==    by 0xF1F105: wasmer_wasi::state::WasiFs::create_fd (mod.rs:1274)
==29872==    by 0xF14A7A: wasmer_wasi::state::WasiFs::new_init (mod.rs:457)
==29872== 
==29872== 512 bytes in 1 blocks are still reachable in loss record 93 of 107
==29872==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x2074F09: aligned_malloc (alloc.rs:95)
==29872==    by 0x2074F09: alloc (alloc.rs:22)
==29872==    by 0x2074F09: __rdl_alloc (alloc.rs:304)
==29872==    by 0x121ED4B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x121F5F3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x121610A: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==29872==    by 0x11D30F0: alloc::raw_vec::RawVec<T,A>::grow_amortized (raw_vec.rs:429)
==29872==    by 0x11D15E3: alloc::raw_vec::RawVec<T,A>::try_reserve (raw_vec.rs:310)
==29872==    by 0x11D5242: alloc::raw_vec::RawVec<T,A>::reserve (raw_vec.rs:300)
==29872==    by 0x11AE168: alloc::vec::Vec<T>::reserve (vec.rs:505)
==29872==    by 0x11AF717: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2136)
==29872==    by 0x11AFFF1: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29872== 
==29872== 640 bytes in 1 blocks are still reachable in loss record 94 of 107
==29872==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x2074F09: aligned_malloc (alloc.rs:95)
==29872==    by 0x2074F09: alloc (alloc.rs:22)
==29872==    by 0x2074F09: __rdl_alloc (alloc.rs:304)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1187645: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0x118EDD7: rayon_core::registry::Registry::new (registry.rs:238)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872== 
==29872== 640 bytes in 1 blocks are still reachable in loss record 95 of 107
==29872==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x2074F09: aligned_malloc (alloc.rs:95)
==29872==    by 0x2074F09: alloc (alloc.rs:22)
==29872==    by 0x2074F09: __rdl_alloc (alloc.rs:304)
==29872==    by 0x11FAC4B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11FB193: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11FABAC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x11F6E25: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0x11F7369: <crossbeam_epoch::collector::Collector as core::default::Default>::default (collector.rs:32)
==29872==    by 0x11F7396: crossbeam_epoch::collector::Collector::new (collector.rs:40)
==29872==    by 0x11F5889: __static_ref_initialize (default.rs:13)
==29872==    by 0x11F5889: core::ops::function::FnOnce::call_once (function.rs:233)
==29872==    by 0x11F56F5: lazy_static::lazy::Lazy<T>::get::{{closure}} (inline_lazy.rs:31)
==29872==    by 0x11F69B4: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872== 
==29872== 808 bytes in 1 blocks are still reachable in loss record 96 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1C76D9B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1C76EC3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1C76CFC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1C850D0: new<alloc::collections::btree::node::LeafNode<usize, wasmer_engine::trap::frame_info::ModuleInfoFrameInfo>> (boxed.rs:177)
==29872==    by 0x1C850D0: alloc::collections::btree::node::Root<K,V>::new_leaf (node.rs:157)
==29872==    by 0x1C6B839: core::ops::function::FnOnce::call_once (function.rs:233)
==29872==    by 0x1C7811C: core::option::Option<T>::get_or_insert_with (option.rs:853)
==29872==    by 0x1C9D9DE: alloc::collections::btree::map::BTreeMap<K,V>::ensure_root_is_owned (map.rs:2324)
==29872==    by 0x1C9DDB1: alloc::collections::btree::map::BTreeMap<K,V>::entry (map.rs:1061)
==29872==    by 0x1C9EC29: alloc::collections::btree::map::BTreeMap<K,V>::insert (map.rs:829)
==29872==    by 0x1C6695F: wasmer_engine::trap::frame_info::register (frame_info.rs:281)
==29872==    by 0x102F744: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (artifact.rs:275)
==29872== 
==29872== 896 bytes in 1 blocks are indirectly lost in loss record 97 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1F20D6B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1F21513: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x1F1FE3A: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==29872==    by 0xDDA044: alloc::raw_vec::RawVec<T,A>::grow_exact (raw_vec.rs:448)
==29872==    by 0xDE1F13: alloc::raw_vec::RawVec<T,A>::try_reserve_exact (raw_vec.rs:347)
==29872==    by 0xDDD5F2: alloc::raw_vec::RawVec<T,A>::reserve_exact (raw_vec.rs:334)
==29872==    by 0xE3BAE8: alloc::vec::Vec<T>::reserve_exact (vec.rs:530)
==29872==    by 0xEB3B19: generational_arena::Arena<T>::reserve (lib.rs:804)
==29872==    by 0xEB31B8: generational_arena::Arena<T>::with_capacity (lib.rs:282)
==29872==    by 0xEB3562: generational_arena::Arena<T>::new (lib.rs:252)
==29872==    by 0xF1475F: wasmer_wasi::state::WasiFs::new_init (mod.rs:422)
==29872== 
==29872== 1,152 bytes in 4 blocks are possibly lost in loss record 98 of 107
==29872==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==29872==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==29872==    by 0x5252227: allocate_stack (allocatestack.c:627)
==29872==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==29872==    by 0x2084B1C: std::sys::unix::thread::Thread::new (thread.rs:66)
==29872==    by 0x11BE0AB: std::thread::Builder::spawn_unchecked (mod.rs:492)
==29872==    by 0x11BF2DB: std::thread::Builder::spawn (mod.rs:386)
==29872==    by 0x118E0F8: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29872==    by 0x118F1DF: rayon_core::registry::Registry::new (registry.rs:260)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872== 
==29872== 1,520 bytes in 1 blocks are still reachable in loss record 99 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x119D431: new<crossbeam_deque::deque::Block<rayon_core::job::JobRef>> (boxed.rs:177)
==29872==    by 0x119D431: <crossbeam_deque::deque::Injector<T> as core::default::Default>::default (deque.rs:1187)
==29872==    by 0x119BCBD: crossbeam_deque::deque::Injector<T>::new (deque.rs:1213)
==29872==    by 0x118EBA3: rayon_core::registry::Registry::new (registry.rs:242)
==29872==    by 0x118E36D: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29872==    by 0x118E6F9: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29872==    by 0x11B1BC3: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29872==    by 0x206FB49: std::sync::once::Once::call_inner (once.rs:416)
==29872==    by 0x11B1B40: std::sync::once::Once::call_once (once.rs:264)
==29872== 
==29872== 1,536 bytes in 4 blocks are still reachable in loss record 100 of 107
==29872==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x2074F09: aligned_malloc (alloc.rs:95)
==29872==    by 0x2074F09: alloc (alloc.rs:22)
==29872==    by 0x2074F09: __rdl_alloc (alloc.rs:304)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x1187465: alloc::sync::Arc<T>::new (sync.rs:324)
==29872==    by 0x119B7ED: crossbeam_deque::deque::Worker<T>::new_lifo (deque.rs:250)
==29872==    by 0x118F65E: rayon_core::registry::Registry::new::{{closure}} (registry.rs:229)
==29872==    by 0x11C836C: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872==    by 0x1198FBD: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29872==    by 0x11C9BC3: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29872== 
==29872== 1,840 bytes in 1 blocks are indirectly lost in loss record 101 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1E46BEB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1E600C0: hashbrown::raw::RawTable<T>::new_uninitialized (mod.rs:383)
==29872==    by 0x1E60958: hashbrown::raw::RawTable<T>::try_with_capacity (mod.rs:407)
==29872==    by 0x1E655C2: hashbrown::raw::RawTable<T>::resize (mod.rs:788)
==29872==    by 0x1E5BBCC: hashbrown::raw::RawTable<T>::reserve_rehash (mod.rs:669)
==29872==    by 0x1E65DA6: hashbrown::raw::RawTable<T>::reserve (mod.rs:626)
==29872==    by 0x1E0BEE1: hashbrown::map::HashMap<K,V,S>::reserve (map.rs:583)
==29872==    by 0x1E0A744: hashbrown::rustc_entry::<impl hashbrown::map::HashMap<K,V,S>>::rustc_entry (rustc_entry.rs:45)
==29872==    by 0x1E22C42: std::collections::hash::map::HashMap<K,V,S>::entry (map.rs:670)
==29872==    by 0x1E55F6B: wasmer_vm::sig_registry::SignatureRegistry::register (sig_registry.rs:46)
==29872==    by 0x10505F9: wasmer_engine_jit::artifact::JITArtifact::from_parts::{{closure}} (artifact.rs:189)
==29872== 
==29872== 1,840 bytes in 1 blocks are indirectly lost in loss record 102 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x1E46BEB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x1E5F9C0: hashbrown::raw::RawTable<T>::new_uninitialized (mod.rs:383)
==29872==    by 0x1E60FF8: hashbrown::raw::RawTable<T>::try_with_capacity (mod.rs:407)
==29872==    by 0x1E644E2: hashbrown::raw::RawTable<T>::resize (mod.rs:788)
==29872==    by 0x1E5BE0C: hashbrown::raw::RawTable<T>::reserve_rehash (mod.rs:669)
==29872==    by 0x1E65D26: hashbrown::raw::RawTable<T>::reserve (mod.rs:626)
==29872==    by 0x1E64035: hashbrown::raw::RawTable<T>::insert (mod.rs:841)
==29872==    by 0x1E0B8FD: hashbrown::map::HashMap<K,V,S>::insert (map.rs:855)
==29872==    by 0x1E22CC2: std::collections::hash::map::HashMap<K,V,S>::insert (map.rs:819)
==29872==    by 0x1E56413: wasmer_vm::sig_registry::SignatureRegistry::register (sig_registry.rs:58)
==29872==    by 0x10505F9: wasmer_engine_jit::artifact::JITArtifact::from_parts::{{closure}} (artifact.rs:189)
==29872== 
==29872== 2,008 bytes in 1 blocks are still reachable in loss record 103 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11FAC4B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11FB193: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11FABAC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x11F3569: new<crossbeam_epoch::sync::queue::Node<crossbeam_epoch::internal::SealedBag>> (boxed.rs:177)
==29872==    by 0x11F3569: <T as crossbeam_epoch::atomic::Pointable>::init (atomic.rs:188)
==29872==    by 0x11F2F15: crossbeam_epoch::atomic::Owned<T>::init (atomic.rs:861)
==29872==    by 0x11F2E65: crossbeam_epoch::atomic::Owned<T>::new (atomic.rs:846)
==29872==    by 0x11F376B: crossbeam_epoch::sync::queue::Queue<T>::new (queue.rs:50)
==29872==    by 0x11EEB63: crossbeam_epoch::internal::Global::new (internal.rs:254)
==29872==    by 0x11F735B: <crossbeam_epoch::collector::Collector as core::default::Default>::default (collector.rs:32)
==29872==    by 0x11F7396: crossbeam_epoch::collector::Collector::new (collector.rs:40)
==29872==    by 0x11F5889: __static_ref_initialize (default.rs:13)
==29872==    by 0x11F5889: core::ops::function::FnOnce::call_once (function.rs:233)
==29872== 
==29872== 4,096 bytes in 4 blocks are still reachable in loss record 104 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11D139D: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==29872==    by 0x11D4610: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==29872==    by 0x11D078E: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==29872==    by 0x11ABF9E: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==29872==    by 0x119A200: crossbeam_deque::deque::Buffer<T>::alloc (deque.rs:45)
==29872==    by 0x119B6D9: crossbeam_deque::deque::Worker<T>::new_lifo (deque.rs:248)
==29872==    by 0x118F65E: rayon_core::registry::Registry::new::{{closure}} (registry.rs:229)
==29872==    by 0x11C836C: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29872==    by 0x1198FBD: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29872== 
==29872== 6,080 bytes in 4 blocks are still reachable in loss record 105 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11B7BAB: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11B89D3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11B7B0C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x119D431: new<crossbeam_deque::deque::Block<rayon_core::job::JobRef>> (boxed.rs:177)
==29872==    by 0x119D431: <crossbeam_deque::deque::Injector<T> as core::default::Default>::default (deque.rs:1187)
==29872==    by 0x119BCBD: crossbeam_deque::deque::Injector<T>::new (deque.rs:1213)
==29872==    by 0x118923C: rayon_core::job::JobFifo::new (job.rs:193)
==29872==    by 0x11915FA: rayon_core::registry::main_loop (registry.rs:806)
==29872==    by 0x118DC2F: rayon_core::registry::ThreadBuilder::run (registry.rs:56)
==29872==    by 0x118E2E0: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn::{{closure}} (registry.rs:101)
==29872==    by 0x11C6B8F: std::sys_common::backtrace::__rust_begin_short_backtrace (backtrace.rs:130)
==29872==    by 0x11BF160: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}} (mod.rs:475)
==29872== 
==29872== 6,669 (16 direct, 6,653 indirect) bytes in 1 blocks are definitely lost in loss record 106 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0xE1BE5B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0xE1BF83: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0xE1BDBC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0xE9AF36: new<wasmer_wasi::WasiEnv> (boxed.rs:177)
==29872==    by 0xE9AF36: wasmer::externals::function::Function::new_native_with_env (function.rs:238)
==29872==    by 0xE67000: wasmer_wasi::generate_import_object_snapshot0 (lib.rs:213)
==29872==    by 0xE66BBC: wasmer_wasi::generate_import_object_from_env (lib.rs:195)
==29872==    by 0xE668D3: wasmer_wasi::WasiEnv::import_object (lib.rs:149)
==29872==    by 0xCAAD6B: wasi::main (wasi.rs:51)
==29872==    by 0xCAE22A: std::rt::lang_start::{{closure}} (rt.rs:67)
==29872==    by 0x20764A2: {{closure}} (rt.rs:52)
==29872==    by 0x20764A2: do_call<closure-0,i32> (panicking.rs:348)
==29872==    by 0x20764A2: try<i32,closure-0> (panicking.rs:325)
==29872==    by 0x20764A2: catch_unwind<closure-0,i32> (panic.rs:394)
==29872==    by 0x20764A2: std::rt::lang_start_internal (rt.rs:51)
==29872==    by 0xCAE206: std::rt::lang_start (rt.rs:67)
==29872== 
==29872== 8,160 bytes in 4 blocks are still reachable in loss record 107 of 107
==29872==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29872==    by 0x11FAC4B: alloc::alloc::alloc (alloc.rs:80)
==29872==    by 0x11FB193: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29872==    by 0x11FABAC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29872==    by 0x11EE599: new<crossbeam_epoch::internal::Local> (boxed.rs:177)
==29872==    by 0x11EE599: <T as crossbeam_epoch::atomic::Pointable>::init (atomic.rs:188)
==29872==    by 0x11F2EB5: crossbeam_epoch::atomic::Owned<T>::init (atomic.rs:861)
==29872==    by 0x11F2E15: crossbeam_epoch::atomic::Owned<T>::new (atomic.rs:846)
==29872==    by 0x11EF402: crossbeam_epoch::internal::Local::register (internal.rs:392)
==29872==    by 0x11F73BD: crossbeam_epoch::collector::Collector::register (collector.rs:45)
==29872==    by 0x11F61A3: crossbeam_epoch::default::HANDLE::__init (default.rs:18)
==29872==    by 0x11F5868: core::ops::function::FnOnce::call_once (function.rs:233)
==29872==    by 0x11FB33B: std::thread::local::lazy::LazyKeyInner<T>::initialize (local.rs:289)
==29872== 
==29872== LEAK SUMMARY:
==29872==    definitely lost: 720 bytes in 45 blocks
==29872==    indirectly lost: 6,653 bytes in 70 blocks
==29872==      possibly lost: 1,152 bytes in 4 blocks
==29872==    still reachable: 29,360 bytes in 91 blocks
==29872==         suppressed: 0 bytes in 0 blocks
==29872== 
==29872== For counts of detected and suppressed errors, rerun with: -v
==29872== ERROR SUMMARY: 46 errors from 46 contexts (suppressed: 0 from 0)
